File: vortexdbg-api/src/main/java/com/vortexdbg/arm/context/AbstractRegisterContext.java
package com.vortexdbg.arm.context;

import com.vortexdbg.pointer.VortexdbgPointer;

public abstract class AbstractRegisterContext implements RegisterContext {

    @Override
    public final int getIntArg(int index) {
        return (int) getLongArg(index);
    }

    @Override
    public final long getLongArg(int index) {
        VortexdbgPointer pointer = getPointerArg(index);
        return pointer == null ? 0 : pointer.peer;
    }

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/arm/context/AbstractRegisterContext.kt
package com.vortexdbg.arm.context

import com.vortexdbg.pointer.VortexdbgPointer

abstract class AbstractRegisterContext : RegisterContext {

    final override fun getIntArg(index: Int): Int {
        return getLongArg(index).toInt()
    }

    final override fun getLongArg(index: Int): Long {
        val pointer: VortexdbgPointer? = getPointerArg(index)
        return if (pointer == null) 0 else pointer.peer
    }

}
